ALEC-228: Create Quick Start section (#97)
OpenNMS/alec: docs/modules/engines/pages/introduction.adoc

= Correlation Engines

Correlation engines are the underlying processes that drive ALEC.

> Two engines are implemented by default: xref:clustering.adoc[Clustering] and xref:deeplearning.adoc[Deep Learning].

Reviewer: ```suggestion
Two engines are implemented by default: xref:clustering.adoc[clustering] and xref:deeplearning.adoc[deep learning].
```

> Both extend the cluster engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).

Reviewer: ```suggestion
Both extend the correlation engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).
```

Author: We may want to talk to the team about the naming here, since the underlying cluster engine is different from the cluster*ing* engine. I'll note it for tomorrow.

> == Cluster engine

Reviewer: correlation engine?

> The cluster correlation engine builds a network topology graph.

Reviewer: ```suggestion
The correlation engine builds a network topology graph.
```

> The graph is based on network inventory information provided by the datasource, and is persisted in local memory.

Reviewer: ```suggestion
The graph is based on network inventory information the datasource provides, and is persisted in local memory.
```

Inventory objects are added to the graph as vertices, and relationships among them are added as edges.
Alarms are then attached to the vertex of the inventory object that they reference.

> On every refresh interval (which defaults to 30 seconds), ALEC runs a clustering algorithm against all alarms on the graph.

Reviewer: I assume this interval is set in the OpenNMS core, so probably no need to document, but let's verify.

This process outputs a list of alarm clusters.
The engine then maps clusters to situations, and sends that situation data back to the datasource.

> The underlying cluster engine is a generic implementation.

Reviewer: ```suggestion
The underlying correlation engine is a generic implementation.
```
